src/bin/cql-stress-cassandra-stress/settings/param/mod.rs

```diff
@@ -76,7 +76,36 @@ impl<P: Param> ParamWrapperImpl<P> {
     }
 }
 
-impl<P: Param> ParamWrapperImpl<P> {
+/// There is only one reason this trait is introduced:
+/// - [super::ParamsParser], as well as [super::ParamsGroup] don't care about the
+///   specific type of the parameter that is being wrapped. They hold the parameters in
+///   some collections (std::Vector) and use a common interface of the [Param].
+/// - On the other hand, the values returned to the user depend on the specific type of the parameter
+///   For example, [simple_param::SimpleParam] exposes `get()` method which returns a simple value,
+///   when [multi_param::MultiParam] exposes `get_arbitrary` method (returning arbitrary parameters as a map).
+///
+/// The implementation for this trait is written only once (below) for all of the generic types
+/// of [ParamWrapperImpl]. Thanks to this trait, [ParamWrapper]s can be held in a common collection
+/// as [Rc<RefCell<dyn ParamWrapper>>]. At the same time, corresponding handles (e.g. [simple_param::SimpleParamHandle])
+/// can access the specific type, exposing specific methods (e.g. [ParamWrapperImpl<simple_param::SimpleParam>]::get()).
```

Comment:
I think this docstring comment should be more "straight to the point". It should explain in one sentence what the item is, then more specific explanations should follow. The current comment throws in lots of references to other items at the very beginning and seems to assume that the reader already knows about the rest of the code.

Take a look at the section about documenting components from the rustdoc book: https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components . Maybe we don't necessarily need to go as far and provide code examples, but the general idea should be that you start with the most important information and explain more details as the docstring unfolds.

I'm also not sure that all the details about other structs/traits need to be included here. Such comments have a tendency to become obsolete or misleading if you restructure other parts of the code.